src/pixelator/graph/backends/implementations/_igraph.py

```diff
@@ -0,0 +1,536 @@
+                v2["markers"][marker] += 1
+
+        # the type attribute is needed to project (True means second projection
+        # which in out case is upib, the first projection would be upia)
+        dest_pixels = set(edgelist["upib"])
+        graph.vs["type"] = [v["name"] in dest_pixels for v in graph.vs]
+        graph.vs["pixel_type"] = list(
+            map(lambda x: "B" if x else "A", graph.vs["type"])
+        )
+
+        if not use_full_bipartite:
+            logger.debug("Projecting graph on UPIA")
+            # which argument defines which projection to retrieve (first (0) or
+            # second (1))
+            graph = graph.simplify().bipartite_projection(
+                types="type", multiplicity=False, which=0
+            )
+
+        logger.debug("Graph created")
+        return IgraphGraphBackend(
+            graph.simplify() if use_full_bipartite and simplify else graph
+        )
+
+    @staticmethod
+    def from_raw(graph: igraph.Graph) -> "IgraphGraphBackend":
+        """Generate a Graph from an igraph.Graph object.
+
+        :param graph: input igraph to use
+        :return: A pixelator Graph object
+        :rtype: IgraphGraphBackend
+        """
+        return IgraphGraphBackend(graph)
+
+    @property
+    def raw(self):
+        """Get the raw underlying graph representation."""
+        return self._raw
+
+    @property
+    def vs(self):
+        """Get a sequence of the vertices in the Graph instance."""
+        return IgraphBasedVertexSequence(self._raw.vs)
+
+    @property
+    def es(self):
+        """A sequence of the edges in the Graph instance."""
+        return IgraphBasedEdgeSequence(self._raw.es)
+
+    def vcount(self):
+        """Get the total number of vertices in the Graph instance."""
+        return self._raw.vcount()
+
+    def ecount(self):
+        """Get the total number of edges in the Graph instance."""
+        return self._raw.ecount()
+
+    def get_adjacency_sparse(self) -> csr_matrix:
+        """Get the sparse adjacency matrix."""
+        return self._raw.get_adjacency_sparse()
+
+    def connected_components(self) -> VertexClustering:
+        """Get the connected components in the Graph instance."""
+        return IgraphBasedVertexClustering(
+            self._raw.connected_components(), graph=self._raw
+        )
+
+    def community_leiden(
+        self,
+        n_iterations: int = 2,
+        beta: float = 0.01,
+        **kwargs,
+    ) -> VertexClustering:
+        """Run community detection using the Leiden algorithm."""
+        return IgraphBasedVertexClustering(
+            self._raw.community_leiden(
+                objective_function="modularity",
+                n_iterations=n_iterations,
+                beta=beta,
+                **kwargs,
+            ),
+            graph=self._raw,
+        )
+
+    def layout_coordinates(
+        self,
+        layout_algorithm: str = "fruchterman_reingold",
```

Comment:
Maybe we can encode the options in a `typing.Literal` here?

Reply:
I suggest sticking with this for now. But I will revisit this later as I work on the layout stuff in networkx.